### DISCUSSION TITLE
Debounce custom for text inputs

### DISCUSSION THREAD

**Discussion by nfuenmamelt** [KevinVandy/material-react-table#854] [Ideas]
Hey guys I'm trying to figure out ho to increase the debounce timer for any change in the inputs column filter, I added my custom debounce in my custom handleColumnFiltersChange, but it affects too multi-select filters, I don't want that.

I was reviewing the code and I saw you are doing this on these lines https://github.com/KevinVandy/material-react-table/blob/v2/packages/material-react-table/src/inputs/MRT_FilterTextField.tsx#L136.

But you put a constant value for the debounce, Can you add a prop to send our custom debounce value?

**Comment by MistyKuu** [comment]
I had similar issue. I wanted to have an apply filter button. Hardcoded debouncing value in the library was causing a bug with missing values when user entered the values and clicked the button immediately. I had to manually assign onChangeCapture event to textFieldFilterProps and disable the buttons for 500ms. Please add an option to configure debounce (or disable it)
